src/mafiasi_link_shortener/settings.py

```diff
@@ -139,16 +141,17 @@
 OPENID_SCOPE = "openid shortlinks"
 OPENID_CLIENT_ID = env.str("SHORTLINK_OPENID_CLIENT_ID")
 OPENID_CLIENT_SECRET = env.str("SHORTLINK_OPENID_CLIENT_SECRET")
-OPENID_REDIRECT_URI = None
 OPENID_FRONTEND_CLIENT_ID = env.str("SHORTLINK_OPENID_FRONTEND_CLIENT_ID")
 
 # rest framework
 REST_FRAMEWORK = {
     "DEFAULT_AUTHENTICATION_CLASSES": [
         "simple_openid_connect.integrations.djangorestframework.authentication.AccessTokenAuthentication",
+        "rest_framework.authentication.SessionAuthentication",
     ],
     "DEFAULT_PERMISSION_CLASSES": [
         "simple_openid_connect.integrations.djangorestframework.permissions.HasTokenScope",
+        "rest_framework.permissions.IsAuthenticated",
```

Comment:
If I understand the django-rest-framework docs correctly, permissions are ANDed by default 
[see here](https://www.django-rest-framework.org/api-guide/permissions/#how-permissions-are-determined):
> Before running the main body of the view each permission in the list is checked. If any permission check fails, an exceptions.PermissionDenied or exceptions.NotAuthenticated exception will be raised, and the main body of the view will not run.

What we want though is OR chaining.

I don't think it is actually possible to define `DEFAULT_AUTHENTICATION_CLASSES` with OR semantics :(

What we could do would be to set the default permissions to [`IsAdminUser`](https://www.django-rest-framework.org/api-guide/permissions/#isadminuser) (which would be effectively *deny all* for most users) and the overwrite the specific permissions in each view.

Reply:
I think they are already overwritten in every view, so we could probably remove them (or replace them with very strict permissions like you said).

Reply:
Done in af47bf34fa2caa742385d8f12e9104676d06ddcd.